--- jasmine.js.orig	2024-01-20 15:06:20.821078800 +0100
+++ jasmine.js	2024-01-20 15:07:10.360495593 +0100
@@ -1,3 +1,4 @@
+/* jscs:disable */
 /*
 Copyright (c) 2008-2021 Pivotal Labs
 
@@ -319,7 +320,7 @@
   };
 
   j$.isAsymmetricEqualityTester_ = function(obj) {
-    return obj ? j$.isA_('Function', obj.asymmetricMatch) : false;
+    return obj?.asymmetricMatch ? j$.isA_('Function', obj.asymmetricMatch) : false;
   };
 
   j$.getType_ = function(value) {
@@ -581,7 +582,7 @@
    * @return {Boolean}
    */
   j$.isSpy = function(putativeSpy) {
-    if (!putativeSpy) {
+    if (!putativeSpy || !putativeSpy.and) {
       return false;
     }
     return (
@@ -2249,7 +2250,7 @@
     };
 
     this.describe = function(description, specDefinitions) {
-      ensureIsNotNested('describe');
+      //ensureIsNotNested('describe');
       ensureIsFunction(specDefinitions, 'describe');
       var suite = suiteFactory(description);
       if (specDefinitions.length > 0) {
