--- lib/jasmine.js.orig	2020-08-24 22:31:16.866022768 -0700
+++ lib/jasmine.js	2020-08-24 22:31:16.878022764 -0700
@@ -1,3 +1,4 @@
+/* jscs:disable */
 /*
 Copyright (c) 2008-2015 Pivotal Labs
 
@@ -143,7 +144,7 @@
   };
 
   j$.isDomNode = function(obj) {
-    return obj.nodeType > 0;
+    return obj.nodeType && obj.nodeType > 0;
   };
 
   j$.fnNameFor = function(func) {
@@ -206,7 +207,7 @@
   };
 
   j$.isSpy = function(putativeSpy) {
-    if (!putativeSpy) {
+    if (!putativeSpy || !putativeSpy.and) {
       return false;
     }
     return putativeSpy.and instanceof j$.SpyStrategy &&
@@ -2594,7 +2595,7 @@
   };
 
   function isAsymmetric(obj) {
-    return obj && j$.isA_('Function', obj.asymmetricMatch);
+    return obj && obj.asymmetricMatch && j$.isA_('Function', obj.asymmetricMatch);
   }
 
   function asymmetricMatch(a, b) {
